docs/Serverless.md

```diff
@@ -324,3 +325,131 @@ export default async (req, res) => {
     app.server.emit('request', req, res);
 }
 ```
+
+## Google Cloud Functions
+
+You can run Fastify as a Google Cloud Functions using [`serverFactory`](https://www.fastify.io/docs/v1.13.x/Server/#serverfactory) option to pass the cloud function HTTP request to Fastify `handler`.
+
+### Creation of Fastify instance with `serverFactory`
+```js
+const fastify = require("fastify")({
+  serverFactory: (handler, opts) => {
+    const server = require("http").createServer((req, res) => {
+      handler(req, res);
+    });
+    return server;
+  },
+});
```

Comment:
I don't think this part is strictly needed. Is it?

Reply:
I took the example code from [serverFactory documentation](https://www.fastify.io/docs/latest/Server/#serverfactory) to easily suggest a starting point.

I can remove it, leaving only the link to the documentation. Let me know what you consider to be the best option.

Alternatively, I could also change the function to return directly the server like this

```js
const fastify = require("fastify")({
  serverFactory: (handler, opts) => {
    return require("http").createServer((req, res) => {
      handler(req, res);
    });
  },
});
```

Reply:
The original post need this part because the `handler` need to be extracted for `firebase-functions`.

If you plan to advice people using `fastify.server.emit('request', request, reply)`, then a custom `serverFactory` actually do nothing but confusion to user why it is needed.

Reply:
Thank you @climba03003. You are right. The first version of my code uses a handler created with `serverFactory`. 

Then I found an example with `fastify.server.emit('request', request, reply)` and I found it more elegant. :-)

This reveals my not complete understanding of `serverFactory` function... :-D 

In any case... I've tested the script without the `serverFactory` option and it works.

What do you think is a better approach? For example for performance, if it is relevant, is it better to use `serverFactory` or `fastify.server.emit`?

Reply:
Keep the usage of `fastify.server.emit`, let `fastify` handle the `http` creation.

Reply:
Awesome. 

I agree the final result is much more elegant and easy to understand.

Can I ask you where could I find documentation about events that could be emitted to `fastify.server`? Just for my study :-)

Reply:
`fastify.server` is `http.Server`. So, check the official document for `nodejs`.
https://nodejs.org/dist/latest-v16.x/docs/api/http.html#event-request